addons/point_of_sale/static/src/js/Screens/OrderManagementScreen/OrderManagementScreen.js

```diff
@@ -0,0 +1,102 @@
+odoo.define('point_of_sale.OrderManagementScreen', function (require) {
+    'use strict';
+
+    const { useContext } = owl.hooks;
+    const { useListener } = require('web.custom_hooks');
+    const ControlButtonsMixin = require('point_of_sale.ControlButtonsMixin');
+    const NumberBuffer = require('point_of_sale.NumberBuffer');
+    const Registries = require('point_of_sale.Registries');
+    const OrderFetcher = require('point_of_sale.OrderFetcher');
+    const IndependentToOrderScreen = require('point_of_sale.IndependentToOrderScreen');
+    const contexts = require('point_of_sale.PosContext');
+
+    class OrderManagementScreen extends ControlButtonsMixin(IndependentToOrderScreen) {
+        constructor() {
+            super(...arguments);
+            useListener('close-screen', this.close);
+            useListener('set-numpad-mode', this._setNumpadMode);
+            useListener('click-order', this._onClickOrder);
+            useListener('next-page', this._onNextPage);
+            useListener('prev-page', this._onPrevPage);
+            useListener('search', this._onSearch);
+            NumberBuffer.use({
+                nonKeyboardInputEvent: 'numpad-click-input',
+                useWithBarcode: true,
+            });
+            this.numpadMode = 'quantity';
+            OrderFetcher.setComponent(this);
+            OrderFetcher.setConfigId(this.env.pos.config_id);
+            this.orderManagementContext = useContext(contexts.orderManagement);
+        }
+        mounted() {
+            OrderFetcher.on('update', this, this.render);
+            this.env.pos.get('orders').on('add remove', this.render, this);
+
+            // calculate how many can fit in the screen.
+            // It is based on the height of the header element.
+            // So the result is only accurate if each row is just single line.
+            const flexContainer = this.el.querySelector('.flex-container');
+            const cpEl = this.el.querySelector('.control-panel');
+            const headerEl = this.el.querySelector('.order-row.header');
+            const val = Math.trunc(
```

Comment:
I am not sure you need to do this, because on phone, you'll probably have a very small number of orders. Maybe you could set the limit to 80, which is the default limit in the web client of Odoo

Reply:
It is in the spec to 'Display as many orders as the screen can hold' so I did that logic.